### ISSUE TITLE
save to 0600 permission

### ISSUE THREAD

**Issue by johnwmail** [jamessan/vim-gnupg#68]
Would you please add the feature, let vim-gnupg plugin save file at 0600 permission?

I dont want let other people read my secret file, even it is encrypted one.

Thank you.

**Comment by jamessan** [comment]
What about doing this?

```vim
autocmd BufWritePost * if get(b:, 'GPGEncrypted', 0) | call setfperm(expand('%'), 'rw-------') | endif
```

**Comment by johnwmail** [comment]
Hi jamessan,
I want to test it, but how to use it?
(sorry, I don't know vim plugin)

**Comment by jamessan** [comment]
This would go in your `~/.vimrc` file, although it needs a couple of safe guards to make sure it behaves properly if your vimrc happens to be sourced again after startup.  The general pattern for safely defining autocommands is to put them in a group, and always clear the group's autocommands before defining more.

```vim
augroup johnwmail
    autocommand!
    autocommand BufWritePost * if getbufvar(expand('<abuf>', 'b:GPGEncrypted', 0)) | call setfperm(expand('<abuf>'), 'rw-------') | endif
augroup END
```

Alternatively, instead of having a `BufWritePost` autocommand that triggers for every buffer, it could be defined only when you successfully open a file with gnupg.  vim-gnupg's User autocommand can be used to help with that.

```vim
augroup johnwmail
    autocommand!
    autocommand User GnuPG autocommand BufWritePost <buffer=abuf> call setfperm(expand('<abuf>'), 'rw-------')
augroup END
```

When vim-gnupg has opened a buffer, it triggers the `User` autocommand with the `GnuPG` pattern.  The above autocommand will then be run, and define a `BufWritePost` autocommand for that specific buffer.

**Comment by johnwmail** [comment]
Sorry, I only see your #3 comment, after I edited my #2 comment.
Please review my #2 comment.
Thanks.

**Comment by jamessan** [comment]
Yeah, I see that now.  That's a trickier issue to solve, since the filename the auto-command is triggering is actually the filename minus the encrypted extension.  There isn't an easy way to get the real filename.

I guess the most robust way to solve it is to have the plugin do it.

**Comment by johnwmail** [comment]
Hi, after added
--------------------------------
augroup my.email@gmail.com
    autocmd!
    autocmd User GnuPG autocmd BufWritePost <buffer=abuf> * call setfperm(expand('<afile>'), 'rw-------')
augroup END
--------------------------------
to ~/.vimrc, I got this error message
--------------------------------
Error detected while processing /home/john/.vimrc:
line  106:
E492: Not an editor command: ^Iautocommand!
line  107:
E492: Not an editor command: ^Iautocommand User GnuPG autocommand BufWritePost <buffer=abuf> call setfperm(expand('<abuf>'), 'rw-------')
Press ENTER or type command to continue
--------------------------------


And if add below to ~/.vimrc
------------------------------------
augroup johnwmail
    autocmd!
    autocmd BufWritePost * if getbufvar(expand('<abuf>'), 'b:GPGEncrypted', 0)) | call setfperm(expand('<afile>'), 'rw-------') | endif
augroup END
------------------------------------
No error for opening vim, but got error when save file
the error message only show a sec, then disappear , I can not to read it

**Comment by johnwmail** [comment]
Reply to #comment 4

Ok, thank you.
